### PR TITLE
KYLIN-4936 Exactly aggregation can't transform to project

### PR DESCRIPTION
**Pull request by zzcclp**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Kylin?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on [Kylin's jira](https://issues.apache.org/jira/browse/KYLIN), and have described the bug/feature there in detail
- [x] Commit messages in my PR start with the related jira ID, like "KYLIN-0000 Make Kylin project open-source"
- [x] Compiling and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] If this change need a document change, I will prepare another pr against the `document` branch
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at user@kylin or dev@kylin by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/kylin/query/runtime/plans/AggregatePlan.scala [hunk 1]
```diff
@@ -115,7 +137,7 @@ object AggregatePlan extends LogEx {
         val aggName = SchemaProcessor.replaceToAggravateSchemaName(index, funcName, hash, argNames: _*)
         if (funcName == FunctionDesc.FUNC_COUNT_DISTINCT) {
```
Comment (zhengshengjun) [inline kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/kylin/query/runtime/plans/AggregatePlan.scala:138]: should we replace "funcName == FunctionDesc.FUNC_COUNT_DISTINCT" with call.getFunc.isCountDistinct  ?

File: kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/kylin/query/runtime/plans/AggregatePlan.scala [hunk 2]
```diff
@@ -53,10 +53,32 @@ object AggregatePlan extends LogEx {
...
+      val hash = System.identityHashCode(rel).toString
+      val aggCols = rel.getRewriteAggCalls.asScala.zipWithIndex.map {
+        case (call: KylinAggregateCall, index: Int)
+          if OLAPAggregateRel.getAggrFuncName(call).equals("COUNT_DISTINCT") =>
+          val dataType = call.getFunc.getReturnDataType
+          val funcName = OLAPAggregateRel.getAggrFuncName(call)
+          val argNames = call.getArgList.asScala.map(dataFrame.schema.names.apply(_))
+          val columnName = argNames.map(col)
+          val aggName = SchemaProcessor.replaceToAggravateSchemaName(index, funcName, hash, argNames: _*)
+          if (dataType.getName == "hllc") {
```
Comment (zhengshengjun) [inline kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/kylin/query/runtime/plans/AggregatePlan.scala:68]: What about adding a 'isHLLCountDistinct' method in FunctionDesc, sometimes I have trouble judging whether this count_distinct function is HLL or precise count distinct  ==

**Comment by zhengshengjun** [comment]
Thanks for solving this problem we've met. I have tested and it perfectly addressed our problem.

File: kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/kylin/query/runtime/plans/AggregatePlan.scala (hunk 1 above)
Comment (zzcclp) [reply kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/kylin/query/runtime/plans/AggregatePlan.scala:138]: Good suggestion. Done

File: kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/kylin/query/runtime/plans/AggregatePlan.scala (hunk 2 above)
Comment (zzcclp) [reply kylin-spark-project/kylin-spark-query/src/main/scala/org/apache/kylin/query/runtime/plans/AggregatePlan.scala:68]: Done

**Review by hit-lacus — APPROVED** [review]
LGTM

**Pull request merged**